client/blocks/login/index.jsx

```diff
@@ -135,10 +137,24 @@ class Login extends Component {
 		// Redirects to / if no redirect url is available
 		const url = redirectTo ? redirectTo : window.location.origin;
 
-		// user data is persisted in localstorage at `lib/user/user` line 157
-		// therefor we need to reset it before we redirect, otherwise we'll get
-		// mixed data from old and new user
-		user.clear( () => ( window.location.href = url ) );
+		// User data is persisted in localstorage at `lib/user/user` line 157.
+		// Only clear the data if a user is currently set, otherwise keep the
+		// logged out state around so that it can be used in signup.
+		if ( user.get() ) {
+			user.clear( () => {
+				window.location.href = url;
+			} );
+		} else {
+			if ( isArray( this.props.signupProgress ) ) {
+				const lastStep = last(
```

Comment:
No need to change this, just a tip: lodash has a `findLast` method which, in this case, would look something like this:

```js
const lastStep = findLast( this.props.signupProgress,  step => step.stepName !== 'user' );
```

Reply:
Updated!